webapp/src/auth.ts

```diff
@@ -254,17 +286,16 @@ export function hasIdentity(): boolean {
     return experimentEnabled && identityProviders().length > 0;
 }
 
-export function loggedIn(): boolean {
-    if (!hasIdentity()) { return false; }
-    const state = getState();
-    return !!state.user?.id;
+export async function loggedIn(): Promise<boolean> {
+    await authCheck();
+    return loggedInSync();
```

Comment:
@eanders-MS , I've made loggedIn async and it blocks waiting on the initial user profile fetch. There is a sync version too that checks the current state but it seemed like most locations really wanted the async version to avoid race conditions.